include/deal.II/base/lazy.h

```diff
@@ -136,7 +138,8 @@ class Lazy
    */
   template <typename Callable>
   void
-  ensure_initialized(const Callable &creator) const;
+  ensure_initialized(const Callable &creator) const
+    DEAL_II_CXX20_REQUIRES((std::is_invocable_r_v<T, Callable>));
```

Comment:
I will note that there is a `std::invocable` concept: https://en.cppreference.com/w/cpp/concepts/invocable The problem is that there is no concept like `std::invocable_r` that also checks the return type. One could do this with `std::invoke_result` (https://en.cppreference.com/w/cpp/types/result_of), but that's no simpler than this here. In the end, the thing in a `requires` clause just has to be a boolean expression. It doesn't have to be a `concept`.